src/dotnet-uninstall/MacOs/FileSystemExplorer.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Tools.Uninstall.Shared.BundleInfo;
+using Microsoft.DotNet.Tools.Uninstall.Shared.BundleInfo.Versioning;
+
+namespace Microsoft.DotNet.Tools.Uninstall.MacOs
+{
+    internal static class FileSystemExplorer
```

Comment:
https://github.com/dotnet/cli/blob/master/scripts/obtain/uninstall/dotnet-uninstall-pkgs.sh#L22

mac has pkgutil that is similar to Windows uninstaller. We should start from that first instead of directly look into the directory, there might be something left in pkgutil .

Reply:
sorry i think i did not make it clear it earlier